Update test_vim9_class.vim

pr3m-kamble opened the pull request:
Fix test for abstract methods in Vim9 classes

- Added a new test `Test_abstract_method()` in `test_vim9_class.vim`.
- This test defines an abstract class `A` with an abstract method `String()`, and a subclass `B` that implements the method.
- Verifies that calling the method on an instance of `B` works as expected without raising `E1027: Missing return statement`.
- Ensures the abstract method is handled correctly during compilation.

pr3m-kamble committed:
Update test_vim9_class.vim

Fix test for abstract methods in Vim9 classes

- Added a new test `Test_abstract_method()` in `test_vim9_class.vim`.
- This test defines an abstract class `A` with an abstract method `String()`, and a subclass `B` that implements the method.
- Verifies that calling the method on an instance of `B` works as expected without raising `E1027: Missing return statement`.
- Ensures the abstract method is handled correctly during compilation.

zeertzjq commented on src/testdir/test_vim9_class.vim:
```diff
@@ -3,6 +3,30 @@
 
+def Test_abstract_method()
+    vim9script
+
```
You don't need to use `vim9script` in a `:def` function

pr3m-kamble committed:
Update test_vim9_class.vim

 Test abstract method functionality in Vim9 classes.
Removed unnecessary 'vim9script' as it's not required inside a :def function.
 Defines an abstract class A with an abstract method and a subclass B that implements the method.
 Verifies that calling the implemented method on an instance of B works without errors.

pr3m-kamble committed:
Update test_vim9_class.vim

Fix comments

yegappan commented on src/testdir/test_vim9_class.vim:
```diff
@@ -3,6 +3,28 @@
 source check.vim
 import './vim9.vim' as v9
 
+def Test_abstract_method()
```
There is already an existing Test_abstract_method() test function that tests these cases.  What new test is added in this function?  Also, this function should not be first function in this file.  Also this should be added to the existing Test_abstract_method() function.

pr3m-kamble committed:
Update test_vim9_class.vim

Function Removed
